### ISSUE TITLE
Introduce project naming in .hanamirc

### ISSUE THREAD

**Issue by joneslee85** [hanami/hanami#500]
## Actual

Project class/naming are inferred from the current directory name that the codes reside.

In many scenarios, apps are deployed to a strictly conformed folder, such as `/var/www/app` and it is not easy to change this to the project naming. OR in other cases where users want to mix hanami project into their existing ecosystem, for eg: using hanamirb as API backend and want to be hosted under `api` folder instead of the project name.

## Proposed

By introducing a new entry `project_name` in `.hanamirc`, we could offer a consistent naming that idempotent.

**Comment by jodosha** [comment]
@joneslee85 What do you mean by project name? Do you want to store

```
project_name=Bookshelf
```

or

```
project_name=bookshelf
```

Which operations this should facilitate? CLI?

**Comment by davydovanton** [comment]
@jodosha if I understand right, this changes would be helpful only for generation commands in cli part. That's why I have a question: why will this feature be helpful in production environment?

**Comment by hlegius** [comment]
I'd use it for POC projects those turns real projects after approved. I don't see any downside, so :ok_hand:

**Comment by jodosha** [comment]
@hlegius 

> I don't see any downside, so :ok_hand:

If it's useful it should be included, otherwise not. We should be careful about what we release. Each feature, each addition to `.hanamirc` is a maintenance problem for us.

**Comment by joneslee85** [comment]
@jodosha I would go for `bookshelf` because it is close to the current logic of picking up directory name.

Because we would not support nested levels of namespace, I'd like to suggest that even for the app generated with namespace, for eg:

```
hanami new "Namespace::AppName"
#=> generates `project_name=app_name`
```

Thought?

**Comment by jodosha** [comment]
@joneslee85 Gotcha. But still struggling to understand why having `project_name` in `.hanamirc` is useful for us.

**Comment by accuser** [comment]
Is .hanamirc even required in a production environment? The only reason that I could see for using the CLI is to check the available routes in production.

**Comment by joneslee85** [comment]
@jodosha @accuser it is not related nor useful for production. It is more for the dev env. 

I find it useful because the generated code does not rely on the current directory and thus is easier to share with others, ready to be mixed in other existing codebase

**Comment by jodosha** [comment]
@joneslee85 

> I find it useful because the generated code does not rely on the current directory

That's a really good reason. In order to make this investigation complete, can you please link here the logic that implements this "directory to name" transformation? Thank you :+1:

**Comment by lucasallan** [comment]
Can I work on this? @jodosha @joneslee85

**Comment by jodosha** [comment]
@joneslee85 PING :smile:

**Comment by lucasallan** [comment]
Updates?

**Comment by joneslee85** [comment]
@lucasallan would you like to take on this? cheers

**Comment by lucasallan** [comment]
Sorry, for some reasons I completely forgot about it - I'm on it now.

**Comment by jodosha** [comment]
Implemented by #607 and https://github.com/hanami/hanami/commit/f04500e662974d8235d2258d7304e671e30b0ca8
